metadataStore/commands.py

```diff
@@ -1,93 +1,89 @@
 from mongoengine import connect
-import time
-# TODO: Add logger
 
-# TODO: Update docs after refactoring
 
-
-def save_header(event_descriptors, beamline_config, unique_id,
-                scan_id, create_time, **kwargs):
-    """Create a header in metadataStore database backend
+def insert_begin_run(time, beamline_id, beamline_config=None, owner=None,
+                     scan_id=None, custom=None):
+    """ Provide a head for a sequence of events. Entry point for an
+    experiment's run.
 
     Parameters
     ----------
-    scan_id : int
-        scan identifier visible to the user and data analysis.
-        Auto incremented by opyhd
-
-    event_descriptors : list
-        List of EventDescriptor objects the header is going to contain
-
-    beamline_config : metadataStore.database.beamline_config.BeamlineConfig
-        BeamlineConfig object used to capture events given headers
-        and event_descriptors
-
-    unique_id : str
-        String field containing unique hash generated by ophyd
-
-    create_time : float
-        Creation time of a Header
-
-
-    Other Parameters
-    ----------------
-
-    owner : str, optional
-        Specifies the unix user credentials of the user creating the entry
-
-    beamline_id : str, optional
+    time : float
+        The date/time as found at the client side when an event is
+        created.
+    beamline_id: str
         Beamline String identifier. Not unique, just an indicator of
         beamline code for multiple beamline systems
-
-    custom : dict, optional
+    owner: str, optional
+        Specifies the unix user credentials of the user creating the entry
+    scan_id : int, optional
+        Unique scan identifier visible to the user and data analysis
+    beamline_config: metadataStore.odm_temples.BeamlineConfig, optional
+        Foreign key to beamline config corresponding to a given run
+    custom: dict, optional
         Additional parameters that data acquisition code/user wants to
         append to a given header. Name/value pairs
-
+    Returns
+    -------
+    begin_run: mongoengine.Document
+        Inserted mongoengine object
     """
-
     connect(db=database, host=host, port=port)
 
-    datetime_create_time = __convert2datetime(create_time)
+    if beamline_config is None:
+        begin_run = BeginRunEvent(time=time, scan_id=scan_id, owner=owner,
+                                  time_as_datetime=__convert2datetime(time),
+                                  beamline_id=beamline_id, custom=custom)
+    else:
+        begin_run = BeginRunEvent(time=time, scan_id=scan_id, owner=owner,
+                                  time_as_datetime=__convert2datetime(time),
+                                  beamline_id=beamline_id, custom=custom,
+                                  beamline_config=beamline_config.id)
+    # Had to update to an if/else statement because appending a ReferenceField apparently breaks mongoengine(eye roll)
```

Comment:
see beamline_config. Optional foreignkey. You cannot append a Reference field: begin_run.beamline_config = ObjectId is invalid. So if user creates via beamline_config fall to else